h2o-py/tests/pyunit_utils/utilsPY.py

```diff
@@ -3431,4 +3432,192 @@ def random_dataset_numeric_only(nrow, ncol, integerR=100, misFrac=0.01):
+    We added an extra level of missing(NA) column.  Note that we do not deal with NA/missing values here
+
+    :param h2oFrame:
+    :param xlist:
+    :param yresp:
+    :return:
+    """
+    import xgboost as xgb
+
+    pandaFtrain = h2oFrame.as_data_frame(use_pandas=True, header=True)
+    nrows = h2oFrame.nrow
+
+    # need to fix categoricals
+    for cname in enumCols:
+        ctemp = pd.get_dummies(pandaFtrain[cname], prefix=cname, drop_first=False)
+        pandaFtrain.drop([cname], axis=1, inplace=True)
+        pandaFtrain = pd.concat([ctemp, pandaFtrain], axis=1)
+
+    c0= h2oFrame[yresp].asnumeric().as_data_frame(use_pandas=True, header=True)
+    pandaFtrain.drop([yresp], axis=1, inplace=True)
+    pandaF = pd.concat([c0, pandaFtrain], axis=1)
+    pandaF.rename(columns={c0.columns[0]:yresp}, inplace=True)
+    newX = list(pandaFtrain.columns.values)
+    data = pandaF.as_matrix(newX)
+    label = pandaF.as_matrix([yresp])
+
+    return xgb.DMatrix(data=data, label=label)
+
+def summarizeResult_binomial(h2oPredictD, nativePred, h2oTrainTimeD, nativeTrainTime, h2oPredictTimeD,
+                             nativeScoreTime, tolerance=1e-6):
+    '''
+    This method will summarize and compare H2OXGBoost and native XGBoost results for binomial classifiers.
+    This method will summarize and compare H2OXGBoost and native XGBoost results for binomial classifiers.
+
+    :param h2oPredictD:
+    :param nativePred:
+    :param h2oTrainTimeD:
+    :param nativeTrainTime:
+    :param h2oPredictTimeD:
+    :param nativeScoreTime:
+    :return:
+    '''
+    # Result comparison in terms of time
+    print("H2OXGBoost train time is {0}s.  Native XGBoost train time is {1}s.\n  H2OXGBoost scoring time is {2}s."
+          "  Native XGBoost scoring time is {3}s.".format(h2oTrainTimeD/1000.0, nativeTrainTime,
+                                                          h2oPredictTimeD, nativeScoreTime))
+    # Result comparison in terms of actual prediction value between the two
+    colnames = h2oPredictD.names
+    h2oPredictD['predict'] = h2oPredictD['predict'].asnumeric()
+    h2oPredictLocalD = h2oPredictD.as_data_frame(use_pandas=True, header=True)
+
+    # compare prediction probability and they should agree if they use the same seed
+    for ind in range(h2oPredictD.nrow):
+        assert abs(h2oPredictLocalD[colnames[2]][ind]-nativePred[ind])<tolerance, "H2O prediction prob: {0} and native " \
+                                                                         "XGBoost prediction prob: {1}.  They are " \
+                                                                         "very different.".format(h2oPredictLocalD[colnames[2]][ind], nativePred[ind])
+
+def summarizeResult_multinomial(h2oPredictD, nativePred, h2oTrainTimeD, nativeTrainTime, h2oPredictTimeD,
+                                nativeScoreTime, tolerance=1e-6):
```

Comment:
In theory, the tolerance should be chosen to be 1e-10 or more since the two sides are equal.  From most of the algo tests I found, they use 1e-6 and that is what I did.  We can make it better if you want.

Reply:
No need to make it more precise, I was just curious why this one was chosen. I chose 1e-4 in the older tests because that was the point where inequalities started to appear. After our fixes, this is no longer true...